sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/GenericLdapMapper.java

```diff
@@ -15,371 +15,273 @@
-                                    group.getName(),
-                                    config.get(LdapConfigKeys.APP_SOURCE))))
-                    .collect(Collectors.toList());
-          case ADDRESS:
-            List<Attribute> addressAttributeList = new ArrayList<>();
-            if (((Map<String, String>) attributeValue).containsKey("id")
-                && config.get(LdapConfigKeys.ADDRESS_SOURCE) != null) {
-              Attribute addressAttribute =
-                  new Attribute(
-                      attributeName,
-                      String.format(
-                          "%s=%s,%s",
-                          // TODO should be a param
-                          "l",
-                          //
-                          ((Map<String, String>) attributeValue).get("id"),
-                          config.get(LdapConfigKeys.ADDRESS_SOURCE)));
-              addressAttributeList.add(addressAttribute);
-            }
-            return addressAttributeList;
-          case LIST_STRING:
-            return ((List<String>) attributeValue)
-                .stream()
-                    .map(value -> new Attribute(attributeName, value))
-                    .collect(Collectors.toList());
-          default:
-            if ((String) attributeValue != "") {
-              return List.of(
-                  new Attribute(
-                      ldapField.getAnnotation(AttributeLdapName.class).value(),
-                      (String) attributeValue));
-            } else {
-              return List.of();
-            }
+  private static List<Attribute> transformSugoiToAttribute(
+      String type, String ldapAttributeName, Object sugoiValue, Map<String, String> config) {
+    switch (type.toUpperCase()) {
+      case "STRING":
+        if ((String) sugoiValue != "") {
+          return List.of(new Attribute(ldapAttributeName, (String) sugoiValue));
+        } else {
+          return List.of();
         }
-      }
-      return null;
-    } catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException e) {
-      logger.info("Unable to map field " + ldapField.getName());
-      throw new RuntimeException(e);
+      case "ORGANIZATION":
+        return List.of(
+            new Attribute(
+                ldapAttributeName,
+                String.format(
+                    "%s=%s,%s",
+                    // TODO should be a param
+                    "uid",
+                    //
+                    ((Organization) sugoiValue).getIdentifiant(),
+                    config.get(LdapConfigKeys.ORGANIZATION_SOURCE))));
+      case "ADDRESS":
+        if (((Map<String, String>) sugoiValue).containsKey("id")
+            && config.get(LdapConfigKeys.ADDRESS_SOURCE) != null) {
+          return List.of(
+              new Attribute(
+                  ldapAttributeName,
+                  String.format(
+                      "%s=%s,%s",
+                      // TODO should be a param
```

Comment:
attribute identifier of objects could be set in userstorage config as the same time of attributes ?

Reply:
see #393